### PR TITLE
Migration for 'React 15.5'

### LINKED ISSUE #356: Use 'create-react-class' package instead of React.createClass
`React.createClass` has been deprecated and extracted to it's own package since [React v15.5.0](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#new-deprecation-warnings).

The following warning is displayed if that version of React is used:
`Warning: Calendar: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.`

See [Migrating from `React.createClass`](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.createclass) to fix this.

This could be a temporary fix, but I think moving to ES6 classes is more appropriate :-)

### LINKED ISSUE #360: Use 'prop-types' package instead of React.PropTypes
`React.PropTypes` has been extracted to it's own package since [React v15.5.0](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#new-deprecation-warnings).

The following warning is displayed if that version of React is used:
`Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.`

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Closes #356  & closes #360

### CODE REVIEW

File: src/Month.js [hunk 1]
```diff
@@ -16,356 +18,351 @@ import Overlay from 'react-overlays/lib/Overlay';
...
+                onSelectSlot={this.handleSelectSlot}
+
+                eventComponent={components.event}
+                eventWrapperComponent={components.eventWrapper}
+                dateCellWrapper={components.dateCellWrapper}
+            />
+        )
+    },
+
+    renderDateHeading({date, className, ...props}) {
```
Comment (MichaelDeBoey) [inline src/Month.js]: linter error: `'date' is missing in props validation  react/prop-types`

**Comment by MichaelDeBoey** [comment]
I think this can be merged @jquense?

**Comment by jquense** [comment]
yes! sorry I just haven't had a chance to review it and try it out. But thank you so much for putting this together

**Comment by MichaelDeBoey** [comment]
Let me know if you have any remarks :-)

**Comment by MichaelDeBoey** [comment]
@jquense  Any news on this one? :-)

**Comment by jquense** [comment]
actually I can pull in the relevant changes here with what I already have

**Comment by MichaelDeBoey** [comment]
@jquense Only see your remark just now :/
Too bad, 'cause I could have done that in just 5mins for you... :-)

**Pull request closed without merging**